src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ObjectToIdCache.cs

```diff
@@ -171,6 +150,9 @@ private static bool IsPrime(int candidate)
         {
             3, 7, 17, 37, 89, 197, 431, 919, 1931, 4049, 8419, 17519, 36353,
             75431, 156437, 324449, 672827, 1395263, 2893249, 5999471,
+            11998949, 23997907, 47995853, 95991737, 191983481, 383966977, 767933981, 1535867969,
+            2146435069, 0X7FEFFFFF
```

Comment:
@khdang  0X7FEFFFFF is the largest allowable array (that isn't of an element that is a single byte in size), and 2146435069 is the largest prime smaller than it, so having followed the path the previous code would have taken, and when it will then overflow, I use 2146435069 as a primal last-ditch, and 0X7FEFFFFF as the ultimate last-ditch (giving up on the advantages of prime modulos). After that there's nowhere left to go.